demo/api/src/importer/decorators/target-entity.decorator.ts

```diff
@@ -0,0 +1,27 @@
+import { FilterQuery } from "@mikro-orm/core";
+
+import { ImporterEntityClass } from "../entities/base-target.entity";
+import { PipeData } from "../pipes/importer-pipe.type";
+
+export type PreProcessor = (inputData: PipeData) => Record<string, unknown>;
+export type PreProcessorMap = Record<string, PreProcessor>;
+export type GetFindConditionCallback = (pipeData: PipeData, additionalData?: Record<string, unknown>) => FilterQuery<object>;
+interface Options {
+    recordProcessors?: PreProcessorMap;
+    getFindCondition?: GetFindConditionCallback;
+}
+export interface TargetEntityMetadata extends Options {
+    name: string;
+}
+export function TargetEntity(options: Options = {}): ClassDecorator {
```

Comment:
What exactly is a TargetEntity? The target of an importer? Maybe ImportTargetEntity would be a better name then. Also, do all import targets have to be entities?

Reply:
Maybe I'm blind but I can't find where this decorator is used to get entities. In which cases is it necessary to get the classes by this decorator? Don't I always explicitly define which class I want to target?

Regarding naming: If it's not always an Entity and not always the result of an import how about: `PipelineTarget` or `PipelineTargetClass`

Reply:
Its not used to find entities but to provide additional functionality. At this point it's use is not yet visible. I can remove it until required